tests/test_api_validate.py

```diff
@@ -3101,6 +3116,158 @@ def test_00_registrationtoken_with_validity_period(self):
             self.assertEqual("matching 1 tokens, Outside validity period", detail.get("message"))
 
 
+class RegistrationAndPasswordToken(MyApiTestCase):
+
+    def setUp(self):
+        self.setUp_user_realms()
+
+    def test_00_registration_tokens(self):
+        # Registration tokens always do a genkey, even if we do not set it
+        with self.app.test_request_context('/token/init',
+                                           method='POST',
+                                           data={'user': 'cornelius',
+                                                 'type': 'registration'},
+                                           headers={'Authorization': self.at}):
+            res = self.app.full_dispatch_request()
+            self.assertTrue(res.status_code == 200, res)
+            data = res.json
+            detail = data.get("detail")
+            regcode = detail.get("registrationcode")
```

Comment:
we could also check here that the `regcode` has the default length and content

Reply:
will do.

Reply:
Done 
It looks like the default regcode contains special characters. But is so, this was a problem, that already existed earlier and we should recheck this and open another issue for that.

Reply:
But according to
https://github.com/privacyidea/privacyidea/blob/4dff583f3e099dd1479b2454115a83c6700e9f93/privacyidea/lib/tokens/registrationtoken.py#L41
the contents should have been "cn". Where did You encounter special characters?

Reply:
Maybe this is the reason why the `APITokenTestCase.test_15_registration_code()` fails sometimes.